keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,337 @@
+  ```Bash
+    export AZURE_CLIENT_ID="generated-app-ID"
+    export AZURE_CLIENT_SECRET="random-password"
+    export AZURE_TENANT_ID="tenant-ID"
+  ```
+
+* Grant the above mentioned application authorization to perform secret operations on the keyvault:
+    ```Bash
+    az keyvault set-policy --name <your-key-vault-name> --spn $AZURE_CLIENT_ID --secret-permissions backup delete get list set
+    ```
+    > --secret-permissions:
+    > Accepted values: backup, delete, get, list, purge, recover, restore, set
+
+* Use the above mentioned Key Vault name to retreive details of your Vault which also contains your Key Vault URL:
+    ```Bash
+    az keyvault show --name <your-key-vault-name> 
+    ```
+
+#### Create Secret client
+Once you've populated the **AZURE_CLIENT_ID**, **AZURE_CLIENT_SECRET** and **AZURE_TENANT_ID** environment variables and replaced **your-vault-url** with the above returned URI, you can create the [SecretClient](TODO-rst-docs):
+
+```Java
+SecretClient client = SecretClient.builder()
+        .endpoint(<your-vault-url>)
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+```
+> NOTE: For using Asynchronous client use SecretAsyncClient instead of SecretClient
+
+
+## Key concepts
+### Secret
+  A secret is the fundamental resource within Azure KeyVault. From a developer's perspective, Key Vault APIs accept and return secret values as strings. In addition to the secret data, the following attributes may be specified:
+* expires: Identifies the expiration time on or after which the secret data should not be retrieved.
+* notBefore: Identifies the time after which the secret will be active.
+* enabled: Specifies whether the secret data can be retrieved.
+* created: Indicates when this version of the secret was created.
+* updated: Indicates when this version of the secret was updated.
+
+### Secret Client:
+The Secret client performs the interactions with the Azure Key Vault service for getting, setting, updating,deleting, and listing secrets and its versions. An asynchronous and synchronous, SecretClient, client exists in the SDK allowing for selection of a client based on an application's use case. Once you've initialized a SecretClient, you can interact with the primary resource types in Key Vault.
+
+## Examples
+### Sync API
+The following sections provide several code snippets covering some of the most common Azure Key Vault Secret Service tasks, including:
+- [Create a Secret](#create-a-secret)
+- [Retrieve a Secret](#retrieve-a-secret)
+- [Update an existing Secret](#update-an-existing-secret)
+- [Delete a Secret](#delete-a-secret)
+- [List Secrets](#list-secrets)
+
+### Create a Secret
+
+Create a Secret to be stored in the Azure Key Vault.
+- `setSecret` creates a new secret in the key vault. if the secret with name already exists then a new version of the secret is created.
+```Java
+SecretClient secretClient = SecretClient.builder()
```

Comment:
Is it required to have the same instantiation of `secretClient` code in all examples or would it be sufficient to just show how to create a client once at the beginning?

Reply:
It is shown separately at the beginning - I think it's fine either way - we can try to compare how they behave - these snippets should be self-contained as much as possible, I leave it up to your judgement

Reply:
Having it in all examples, allows code snippets to be copied and pasted without compile issues.
User will need to copy snippets from different places otherwise to get started.

Reply:
Yeah, the five extra lines per code sample does bother me a bit too. I would create a first sample that demonstrates creating a client, and then just assume it exists in later code samples.

Reply:
Okay, 
I've updated for the first sample to have client code, 
rest don't